llvm/lib/Target/AIE/AIEEliminateDuplicatePHI.cpp

```diff
@@ -120,26 +120,26 @@ class AIEEliminateDuplicatePHI : public MachineFunctionPass {
   bool processMBB(MachineBasicBlock &MBB, MachineRegisterInfo &MRI,
                   MachineIRBuilder &MIB, GISelObserverWrapper &Observer) {
     bool Changed = false;
-    auto LastPHI = MBB.getFirstNonPHI();
-    for (auto &MI : make_early_inc_range(MBB.phis())) {
-      if (!isPointerTypePHI(MRI, MI))
-        continue;
-
-      // Ensure that next MI is valid
-      auto NextMI = std::next(MachineBasicBlock::instr_iterator(MI));
-      if (NextMI == MBB.instr_end() || NextMI == LastPHI)
+    const auto End = MBB.end();
```

Comment:
We are comparing every pointer PHI node in the block to any other pointer PHI node in that block.  We know that all PHI nodes are consecutive at the start of the block. The transformation only changes operands in other instructions, so iterators stay intact. By writing the loops using basic iterator idiom reveals the symmetry between the two loops. It also avoids using constructs like .phis() and .getFirstNonPhi() that both perform exactly the same iteration as the main loops.